3-0-spring-framework/3-3-0-enable-string-trimming/src/test/java/com/bobocode/TrimmingPostProcessorTest.java

```diff
@@ -0,0 +1,255 @@
+
+        assertTrue(targetValues.isPresent());
+        assertEquals(1, targetValues.get().length);
+        assertEquals(ElementType.PARAMETER, targetValues.get()[0]);
+    }
+
+    @Test
+    @Order(2)
+    @DisplayName("Annotation @Trimmed has retention <Runtime>")
+    void trimmedAnnotationHasRuntimeRetention() {
+        Optional<Class<? extends Annotation>> annotation = findAnnotationInPackage("Trimmed");
+        Optional<Retention> retention = annotation.map(a -> a.getAnnotation(Retention.class))
+                .filter(r -> r.value().equals(RetentionPolicy.RUNTIME));
+
+        assertTrue(retention.isPresent());
+    }
+
+    @Test
+    @Order(3)
+    @DisplayName("Annotation @EnableStringTrimming has target <Type>")
+    void enableStringTrimmingAnnotationHasCorrectTarget() {
+        Optional<Class<? extends Annotation>> annotation = findAnnotationInPackage("EnableStringTrimming");
+        Optional<ElementType[]> targetValues = annotation.map(a -> a.getAnnotation(Target.class)).map(Target::value);
+
+        assertTrue(targetValues.isPresent());
+        assertEquals(1, targetValues.get().length);
+        assertThat(List.of(targetValues.get())).contains(ElementType.TYPE);
+
+    }
+
+    @Test
+    @Order(4)
+    @DisplayName("Annotation @EnableStringTrimming has retention <Runtime>")
+    void enableStringTrimmingAnnotationHasRuntimeRetention() {
+        Optional<Class<? extends Annotation>> annotation = findAnnotationInPackage("EnableStringTrimming");
+        Optional<Retention> retention = annotation.map(a -> a.getAnnotation(Retention.class))
+                .filter(r -> r.value().equals(RetentionPolicy.RUNTIME));
+
+        assertTrue(retention.isPresent());
+    }
+
+    @Test
+    @Order(5)
+    @DisplayName("Annotation @EnableStringTrimming imports StringTrimmingConfiguration")
+    void enableStringTrimmingAnnotationImportsCorrectConfig() {
+        Optional<Class<? extends Annotation>> annotation = findAnnotationInPackage("EnableStringTrimming");
+        List<Class<?>> importValues = Arrays.stream(annotation
+                        .map(a -> a.getAnnotation(Import.class))
+                        .map(Import::value)
+                        .orElseGet(() -> new Class<?>[0]))
+                .toList();
+        List<String> importNames = importValues.stream().map(Class::getSimpleName).collect(toList());
+        assertThat(importNames).contains("StringTrimmingConfiguration");
+    }
+
+    @Test
+    @Order(6)
+    @DisplayName("StringTrimmingConfiguration does not annotated by @Configuration spring annotation, it should be imported to " +
+            "config class by @EnableStringTrimming")
+    void stringTrimmingConfigurationAnnotatedAsConfiguration() {
+        Optional<Class<?>> configClass = findClassInPackage("StringTrimmingConfiguration");
+        assertTrue(configClass.isPresent());
+        assertNull(configClass.get().getDeclaredAnnotation(Configuration.class));
```

Comment:
It could be two separate tests 
```java
        assertTrue(configClass.isPresent());
        assertNull(configClass.get().getDeclaredAnnotation(Configuration.class));
```